test/test_binary_ufuncs.py

```diff
@@ -827,11 +827,7 @@ def do_test(op, a, b):
                 a = torch.tensor(a, device=devices[0])
                 b = torch.tensor(b, device=devices[1])
 
-                if op in (operator.floordiv, torch.floor_divide):
-                    with self.assertWarnsOnceRegex(UserWarning, "floor_divide"):
-                        do_test(op, a, b)
-                else:
-                    do_test(op, a, b)
+                do_test(op, a, b)
```

Comment:
What happened here?

Reply:
Previously  the tensor device check is done in kernel (after emitting the `floor_divide is deprecated` warning). So it gets both warning (deprecated) and error (on different device). 

Now device check is happening at dispatch time, so no warning in this case. (warning will still be tested in other test case where operands on the same device)